### ISSUE TITLE
maxWidth and maxHeight don't respected after browser window resizing

### ISSUE THREAD

**Issue by NewOldMax** [bokuweb/react-rnd#221]
### Overview of the problem

I'm using react-rnd 7.3.0
My **browser** is:  Chrome 63.0.3239.132 (Win 10)
I am sure this issue is **not a duplicate**

### Description
````javascript
render() {
    return (
        <Rnd
           ....
           maxWidth={document.body.clientWidth - 500} // for example
        >
    ...);
}
````
I'm triggering rerender after browser window size change, so I'm changing maxWidth prop, but it doesn't affect maxWidth of `<Rnd>` component, because it calculates only in `onResizeStart` and `onResizeStop` functions.
And if after resizing window you want to resize `<Rnd>` component, it will blink for a moment.

### Steps to Reproduce

1. Create `<Rnd>` component with maxWidth prop that depends on browser window width
2. Resize browser window
3. Check maxWidth prop on `<Rnd>` component

### Expected behavior

Changing maxWidth and maxHeight props should change maxWidth and maxHeight of `<Rnd>` component

### Actual behavior

Changing maxWidth and maxHeight props are not changing maxWidth and maxHeight of `<Rnd>` component

**UPD:** I think it would be nice to have some method like `recalculateBoundaries` which will calculate maxWidth and maxHeight, so user can call it when he need it

**Comment by bokuweb** [comment]
@NewOldMax Thanks for your report. Would you please provide reproduced project in codesandobox?

**Comment by NewOldMax** [comment]
@bokuweb  https://codesandbox.io/s/vvw796q5ry
resize rectangle to maxWidth and after that resize window. You can see in console that `maxWidth` property is changed, but rectangle maxWidth is not changed

**Comment by bokuweb** [comment]
@NewOldMax Thanks for reproduced project :) I've fixed it and published as v8.0.1.
would you please check this version if you get a chance.

http://bokuweb.github.io/react-rnd/stories/?selectedKind=sandbox&selectedStory=body%20size%20apply%20to%20maxwidth&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

**Comment by NewOldMax** [comment]
It works, thanks 🍺
